File: rb/features.rb
# FreeSports SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module FreeSportsFeatures
  def self.make_feature(name)
    case name
    when "base"
      FreeSportsBaseFeature.new
    when "test"
      FreeSportsTestFeature.new
    else
      FreeSportsBaseFeature.new
    end
  end
end

File: py/features.py
# FreeSports SDK feature factory

from feature.base_feature import FreeSportsBaseFeature
from feature.test_feature import FreeSportsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FreeSportsBaseFeature(),
        "test": lambda: FreeSportsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
